apps/gitness/src/framework/hooks/useLogs.ts

```diff
@@ -0,0 +1,59 @@
+import { useEffect, useRef, useState } from 'react'
+import { LivelogLine } from '@harnessio/code-service-client'
+import { ExecutionState } from '../../types'
+
+interface UseLogsReturnType {
+  logs: LivelogLine[]
+}
+
+enum LogsSSEEvent {
+  MESSAGE = 'message'
+}
+
+interface UseLogsInterface {
+  repoPath: string
+  pipelineId: string
+  stageNum: number
+  stepNum: number
+  stepStatus?: ExecutionState
+  executionNum: string
+}
+
+export const useLogs = ({
```

Comment:
should i similarily implement this into the usespacesse ? @vardanbansal-harness

Reply:
It's upto you and not a bad idea actually. `useSpaceSSE` currently handles more than one type of events so we can make a separate `usePREvents` that only handles events for pull requests.